src/gui/midi.cpp

```diff
@@ -106,20 +106,31 @@ struct DB_Midi {
 	struct {
 		uint8_t buf[SYSEX_SIZE];
 		size_t used;
-		Bitu delay;
-		uint32_t start;
+		uint32_t delay; // ms
+		uint32_t start; // ms
 	} sysex;
 	bool available;
 	MidiHandler * handler;
 };
 
 DB_Midi midi;
 
+/* Explanation for this formula can be found in discussion under patch,
+ * that introduced it: https://sourceforge.net/p/dosbox/patches/241/
+ */
+uint32_t delay_in_ms(size_t sysex_bytes_num)
+{
+	constexpr double bytes_per_s = 3125.0;
+	const auto delay = (sysex_bytes_num * 1.25) / bytes_per_s;
+	return static_cast<uint32_t>(delay * 1000. + 2);
```

Comment:
Could cook-out the division by 1,000 and re-multiplication by 1,000:
``` c++
constexpr double bytes_per_ms = 3.125;
const auto delay_ms = (sysex_bytes_num * 1.25) / bytes_per_ms;
return static_cast<uint32_t>(delay_ms) + 2;

Reply:
Used this suggestion and changed variable name a little bit.